File: benchmarks/oopsla25/subjects/432.c
#include <assert.h>
#include <fcntl.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <unistd.h>
#include <stdbool.h>

bool f432(const char *s) {
    if (s[0] == 'b') {
        fprintf(stderr, "Exception\n");
        exit(1);
    }
    
    size_t n = strlen(s);
    if (s[n - 1] == 'a' || s[n - 1] == 'b') {
        if (n > 1) {
            fprintf(stderr, "Exception\n");
            exit(1);
        }
        return true;
    } else {
        assert(n == 1);
        return false;
    }
}

int main(int argc, char *argv[]) {
  char my_string[10240] = {0};
  if (argc == 1) {
    read(fileno(stdin), my_string, 10240);
  } else {
    int fd = open(argv[1], O_RDONLY);
    read(fd, my_string, 10240);
    close(fd);
  }
  bool b = f432(my_string);
  printf("%s\n", b ? "True" : "False");
  return 0;
}

File: tests/oopsla25/432.py
#!/usr/bin/env python3

import sys

def f432(s: str):
  if s[0] == "b":
    raise Exception
  n = len(s)
  if s[n-1] == "a" or s[n-1] == "b":
    if n > 1:
      raise Exception
    return True
  else:
    assert len(s) == 1
    return False
  
if __name__ == '__main__':
  my_string = ""
  if len(sys.argv) == 1:
    my_string = sys.stdin.read()
  else:
    fd = open(sys.argv[1], 'r')
    my_string = fd.read()
    fd.close()
  
  print(f432(my_string))
